lib/tessel/deploy.js

```diff
@@ -436,7 +436,10 @@ actions.tarBundle = function(opts) {
         })
         .bundle(function(error, results) {
           if (error) {
-            return reject(error);
+            return reject({
```

Comment:
A bunch of other places in the code return a string for a warning and an actual Error for a more stern error output. Why not return an Error here?